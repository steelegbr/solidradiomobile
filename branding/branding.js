/**
 * Provides branding functions.
 */

import { DefaultTheme, DarkTheme } from 'react-native-paper';

/**
 * Generates a station theme.
 * @param {bool} darkMode An indication if we are in dark mode.
 * @param {station} station The station this theme is for.
 */

export function generateStationTheme(darkMode, station) {

    let theme = generateTheme(darkMode);
    theme.colors.primary = station.primary_colour;
    theme.colors.accent = station.primary_colour;
    theme.mode = 'adaptive';
    return theme;

}

/**
 * Generates a theme with no station.
 * @param {bool} darkMode An indication if we are in dark mode.
 */

export function generateTheme(darkMode) {

    if (darkMode) {
        return {
            ...DarkTheme,
            roundness: 10,
            colors: {
                ...DarkTheme.colors
            }
        };
    } else {
        return {
            ...DefaultTheme,
            roundness: 10,
            colors: {
                ...DefaultTheme.colors
            }
        };
    }

}
