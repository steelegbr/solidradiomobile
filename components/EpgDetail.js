/**
 * Detailed EPG listing view.
 */

import React from 'react';
import { connect } from 'react-redux';
import { Component } from 'react';
import { ScrollView, Dimensions } from 'react-native';
import { Card, Text, Title, Subheading } from 'react-native-paper';
import { showTimeSlug } from '../epg/timezone';


class EpgDetail extends Component {

    render() {

        const { listing, stationTimezone, userTimezone, day, styles } = this.props;

        const timeSlug = showTimeSlug(listing, day, userTimezone, stationTimezone, true);

        return  (
            <ScrollView
                contentContainerStyle={styles.wrapper}
            >
                <Card
                    style={styles.card}
                >
                    <Card.Cover source={{ uri: listing.image }} />
                    <Card.Content>
                        <Title>{ listing.title }</Title>
                        <Subheading>{ timeSlug }</Subheading>
                        <Text>{ listing.description }</Text>
                    </Card.Content>
                </Card>
            </ScrollView>
        );

    }

}

const mapStateToProps = (state, ownProps) => {

    const currentStation = state.currentStation;

    return {
        listing: ownProps.route.params.listing,
        stationTimezone: state.stations[currentStation].timezone,
        userTimezone: state.timezone,
        day: ownProps.route.params.day,
        styles: {
            card: {
                width: Dimensions.get('window').width * 0.9,
            },
            wrapper: {
                alignItems: 'center'
            }
        }
    };

}

export default connect(mapStateToProps)(EpgDetail);