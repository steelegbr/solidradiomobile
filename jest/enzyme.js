/**
    Solid Radio Mobile App
    Copyright (C) 2020-2021 Marc Steele

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

import 'react-native';
import 'jest-enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

/**
 * Set up Enzyme to mount to DOM, simulate events,
 * and inspect the DOM in tests.
 */
Enzyme.configure({ adapter: new Adapter() });

// JSDOM - required for Enzyme's mount

const { JSDOM } = require('jsdom');
const jsdom = new JSDOM('<!doctype html><html><body></body></html>');
const { window } = jsdom;
global.window = window;
global.document = window.document;