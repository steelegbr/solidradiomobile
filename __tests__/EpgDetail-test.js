/**
 * Tests the EPG details screen.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import EpgDetail from '../components/EpgDetail';
import { Button } from 'react-native-paper';
import { mount } from 'enzyme';
import { generateTheme } from '../branding/branding';

// Redux

const mockStore = configureStore([]);
const testStationName = 'Test Station';

// Tests

it.each`
stationTimezone | userTimezone | title | image | description | day | start
${'Europe/London'} | ${'Europe/London'} | ${"Night Shift"} | ${"https://www.solidradio.co.uk/wp-content/uploads/2019/08/31548926930_f1a1103e5f_o.jpg"} | ${"We donâ€™t bore you with soppy love songs through the night shift. Instead, enjoy a great mix of songs, all night long."} | ${0} | ${"00:00:00"}
${'Europe/London'} | ${'America/Los_Angeles'} | ${"Solid Country"} | ${"https://www.solidradio.co.uk/wp-content/uploads/2020/04/guitar-839168_1920.jpg"} | ${"Join Stephen as he plays a mix of country hits new and old."} | ${6} | ${"10:00:00"}
`('render-show', ({ stationTimezone, userTimezone, title, image, description, day, start }) => {
    // Arrange

    const store = mockStore({
        stations: {
            'Test Station': {
                timeZone: stationTimezone
            }
        },
        currentStation: 'Test Station',
        timezone: userTimezone
    });

    const listing = {
        title: title,
        description: description,
        image: image,
        start: start
    };

    const route = {
        params: {
            listing: listing,
            day: day
        }
    };

    // Act

    const epgRender = renderer.create(
        <Provider store={store}>
            <EpgDetail
                route={route}
            />
        </Provider>
    ).toJSON();

    // Assert

    expect(epgRender).toMatchSnapshot();

});
