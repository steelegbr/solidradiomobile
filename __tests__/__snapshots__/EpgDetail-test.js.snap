// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render-show 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 4,
          "elevation": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.5,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 0.75,
          "width": 675,
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#eeeeee",
                "height": 195,
                "overflow": "hidden",
              },
              Object {
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              undefined,
            ]
          }
        >
          <Image
            siblings={
              Array [
                "withTheme(Card.Cover)",
                "Card.Content",
              ]
            }
            source={
              Object {
                "uri": "https://www.solidradio.co.uk/wp-content/uploads/2019/08/31548926930_f1a1103e5f_o.jpg",
              }
            }
            style={
              Array [
                Object {
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "flex-end",
                  "padding": 16,
                  "resizeMode": "cover",
                  "width": undefined,
                },
                Object {
                  "borderTopLeftRadius": 4,
                  "borderTopRightRadius": 4,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 16,
              },
              Object {
                "paddingBottom": 16,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "500",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 20,
                      "letterSpacing": 0.15,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    undefined,
                  ],
                  Array [
                    Object {
                      "fontSize": 20,
                      "letterSpacing": 0.15,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            Night Shift
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 16,
                      "letterSpacing": 0.5,
                      "lineHeight": 24,
                      "marginVertical": 2,
                    },
                    undefined,
                  ],
                  Array [
                    Object {
                      "fontSize": 16,
                      "letterSpacing": 0.5,
                      "lineHeight": 24,
                      "marginVertical": 2,
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            Mon 00:00
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            We don’t bore you with soppy love songs through the night shift. Instead, enjoy a great mix of songs, all night long.
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "marginBottom": 20,
          "marginTop": 10,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": 675,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "marginLeft": 12,
                "marginRight": -4,
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰁍
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`render-show 2`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 4,
          "elevation": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.5,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 0.75,
          "width": 675,
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#eeeeee",
                "height": 195,
                "overflow": "hidden",
              },
              Object {
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              undefined,
            ]
          }
        >
          <Image
            siblings={
              Array [
                "withTheme(Card.Cover)",
                "Card.Content",
              ]
            }
            source={
              Object {
                "uri": "https://www.solidradio.co.uk/wp-content/uploads/2020/04/guitar-839168_1920.jpg",
              }
            }
            style={
              Array [
                Object {
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "flex-end",
                  "padding": 16,
                  "resizeMode": "cover",
                  "width": undefined,
                },
                Object {
                  "borderTopLeftRadius": 4,
                  "borderTopRightRadius": 4,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 16,
              },
              Object {
                "paddingBottom": 16,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "500",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 20,
                      "letterSpacing": 0.15,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    undefined,
                  ],
                  Array [
                    Object {
                      "fontSize": 20,
                      "letterSpacing": 0.15,
                      "lineHeight": 30,
                      "marginVertical": 2,
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            Solid Country
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 16,
                      "letterSpacing": 0.5,
                      "lineHeight": 24,
                      "marginVertical": 2,
                    },
                    undefined,
                  ],
                  Array [
                    Object {
                      "fontSize": 16,
                      "letterSpacing": 0.5,
                      "lineHeight": 24,
                      "marginVertical": 2,
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            Sun 10:00 [Sun 02:00]
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                undefined,
              ]
            }
          >
            Join Stephen as he plays a mix of country hits new and old.
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "marginBottom": 20,
          "marginTop": 10,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": 675,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "marginLeft": 12,
                "marginRight": -4,
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰁍
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
